tests/integration/Core/Image/ImageStorage/ImageStorageTest.php

```diff
@@ -0,0 +1,200 @@
+
+    /** @var \eZ\Publish\Core\IO\MetadataHandler|\PHPUnit\Framework\MockObject\MockObject */
+    private $imageSizeMetadataHandler;
+
+    /** @var \eZ\Publish\Core\FieldType\Image\PathGenerator|\PHPUnit\Framework\MockObject\MockObject */
+    private $pathGenerator;
+
+    /** @var \eZ\Publish\Core\Base\Utils\DeprecationWarnerInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $deprecationWarner;
+
+    /** @var \eZ\Publish\Core\FieldType\Image\AliasCleanerInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $aliasCleaner;
+
+    /** @var \eZ\Publish\Core\IO\FilePathNormalizerInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $filePathNormalizer;
+
+    /** @var \eZ\Publish\Core\IO\IOServiceInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $ioService;
+
+    /** @var \eZ\Publish\Core\FieldType\Image\ImageStorage */
+    private $storage;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->redecorator = $this->createMock(UrlRedecoratorInterface::class);
+        $this->gateway = new DoctrineStorage($this->redecorator, $this->getDatabaseConnection());
+        $this->imageSizeMetadataHandler = $this->createMock(MetadataHandler::class);
+        $this->pathGenerator = $this->createMock(PathGenerator::class);
+        $this->deprecationWarner = $this->createMock(DeprecationWarnerInterface::class);
+        $this->aliasCleaner = $this->createMock(AliasCleanerInterface::class);
+        $this->filePathNormalizer = $this->createMock(FilePathNormalizerInterface::class);
+        $this->ioService = $this->createMock(IOServiceInterface::class);
+        $this->storage = new ImageStorage(
+            $this->gateway,
+            $this->ioService,
+            $this->pathGenerator,
+            $this->imageSizeMetadataHandler,
+            $this->deprecationWarner,
+            $this->aliasCleaner,
+            $this->filePathNormalizer,
+        );
+    }
+
+    public function testHasFieldData(): void
+    {
+        self::assertTrue($this->storage->hasFieldData());
+    }
+
+    private function getContext(): array
+    {
+        return ['identifier' => 'LegacyStorage'];
+    }
+
+    /**
+     * @dataProvider providerOfFieldData
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\Core\IO\Exception\InvalidBinaryFileIdException
+     */
+    public function testStoreFieldDataDuringCreate(VersionInfo $versionInfo, Field $field): void
+    {
+        $binaryFile = $this->runCommonStoreFieldDataMocks($field);
+
+        $this->redecorator
+            ->expects(self::exactly(3))
+            ->method('redecorateFromSource')
+            ->with($binaryFile->uri)
+            ->willReturn($binaryFile->uri);
+
+        $this->storage->storeFieldData($versionInfo, $field, $this->getContext());
+
+        self::assertEquals(1, $this->gateway->countImageReferences($binaryFile->uri));
```

Comment:
Prefer strict equality checks.
```suggestion
        self::assertSame(1, $this->gateway->countImageReferences($binaryFile->uri));
```